File: src/equalshots.lua
a, b = io.read("*n", "*n")
va, vb = 0, 0
for _ = 1, a do
    c, v = io.read("*n", "*n")
    va = va + c * v
end
for _ = 1, b do
    c, v = io.read("*n", "*n")
    vb = vb + c * v
end
if va == vb then
    print("same")
else
    print("different")
end

File: src/equalshots.py
a, b = [int(d) for d in input().split()]
va, vb = 0, 0
for _ in range(a):
    c, v = [int(d) for d in input().split()]
    va += c * v
for _ in range(b):
    c, v = [int(d) for d in input().split()]
    vb += c * v
print("same" if va == vb else "different")
